lib/manageiq/appliance_console/oidc_authentication.rb

```diff
@@ -85,10 +88,26 @@ def validate_oidc_options
       def derive_introspection_endpoint
         return if options[:oidc_introspection_endpoint].present?
 
-        options[:oidc_introspection_endpoint] = options[:oidc_url].sub(URL_SUFFIX, INTROSPECT_SUFFIX) if options[:oidc_url].match(URL_SUFFIX)
+        options[:oidc_introspection_endpoint] = fetch_introspection_endpoint || compose_introspection_endpoint
         raise INTROSPECT_ENDPOINT_ERROR if options[:oidc_introspection_endpoint].blank?
       end
 
+      def fetch_introspection_endpoint
+        uri = URI.parse(options[:oidc_url])
+        http = Net::HTTP.new(uri.host, uri.port)
+        http.use_ssl = (uri.scheme == "https")
```

Comment:
Will this work with self signed cert in local keycloak setup ? we may need to add http.verify_mode = OpenSSL::SSL::VERIFY_NONE with an option like --insecure

Reply:
Correct. this will not work with self signed cert and I am concerned  it would be safe to use `http.verify_mode = OpenSSL::SSL::VERIFY_NONE`.
Which is why I didn't drop the `compose_introspection_endpoint`.

@abellotti Do you think using  OpenSSL::SSL::VERIFY_NONE is safe?

Reply:
Of course not, default must always be on.    Can't we make it optional for dev setups (i.e. via an option like --insecure or something similar to that) ?